src/node.cpp

```diff
@@ -142,4 +148,29 @@ std::vector<session_key_t> Node::getReadySessions() const {
     return std::move(readySessions);
 }
 
+Status Node::demultiplyOutputs(SessionResults& nodeSessionOutputs) {
+    auto& [metadata, blobMap] = nodeSessionOutputs.begin()->second;
+    auto& [session, blob] = *blobMap.begin();
+    std::vector<NodeSessionMetadata> newSessionMetadatas(metadata.generateSubsessions(getName(), demultiplexCount.value()));
+    auto tensorDesc = blob->getTensorDesc();
+    auto newDims = tensorDesc.getDims();
+    newDims.erase(newDims.begin() + 1);
```

Comment:
What if `newDims.size()` is equal to 1?

Reply:
Config validation should ensure that the size is >= 2. But as a safety measure we could return here additional retCode.